config/confignet/confignet.go

```diff
@@ -27,7 +70,7 @@ type NetAddr struct {
 
 	// Transport to use. Known protocols are "tcp", "tcp4" (IPv4-only), "tcp6" (IPv6-only), "udp", "udp4" (IPv4-only),
 	// "udp6" (IPv6-only), "ip", "ip4" (IPv4-only), "ip6" (IPv6-only), "unix", "unixgram" and "unixpacket".
-	Transport string `mapstructure:"transport"`
+	Transport TransportType `mapstructure:"transport"`
```

Comment:
This is breaking stuff in Contrib (expected).  Do we need to introduce this change as a new field called `TransportType` and follow the deprecation process to get rid of `Transport` and then rename back to `Transport`?  I'd prefer not to do another 3-release cycle change.

Reply:
Looking at this again and taking the new Enumeration expectations into account, I think it is better if the config field name matches the enum name.  I've pushed a commit that adds a new field, `TransportType`, and deprecates `Transport`.  If we can get this merged before 0.94.0 that means we can remove `Transport` after `0.94.0` and mark as stable for the `0.95.0` release.

Reply:
Hmm but that breaks user configurations (config goes from transport to transport_type), not just the API. What do you think?

Reply:
@TylerHelmuth we can put `transport` as the `mapstructure` tag for `TransportType` and put some dummy value on the deprecated field (we tipically have used `-` for unexported fields). Since `TransportType` unmarshals in the same way as `Transport`, this should make it non-breaking for user configurations

Reply:
Oh clever, I'll make that change

Reply:
Done

Reply:
@mx-psi I believe configs won't be broken, but we'll break scenarios [like this](https://github.com/open-telemetry/opentelemetry-collector/actions/runs/7878581831/job/21497142469?pr=9385) where the user is expecting `Transport` to be set instead of `TransportType`.  @bogdandrutu is this the bug you're thinking of?

Reply:
All of this is just making me want to remove `Transport` and do this as a breaking change since this is an infrequently used package and the fix is easy.

Reply:
Ive update the PR to do this as a breaking change.